### PR TITLE
Added Build Workflow for Github Actions

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Added Build Workflow for Github Actions
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | N/A
| Possible impacts? | N/A

Can't check on my fork as Github Actions are not enabled in the repo

### CODE REVIEW

**Review by eternoendless — APPROVED** [review]
LGTM

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
@Progi1984 I fetch your branch and pushed it on my fork.

It fails for a very weird reason 😮 

https://github.com/matks/decimal/runs/1935660154

```
# Install dependencies

0s
Run composer install --no-interaction
  composer install --no-interaction
  shell: /bin/bash -e {0}
Composer could not find a composer.json file in /home/runner/work/decimal/decimal
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
Error: Process completed with exit code 1.
```

File: .github/workflows/php.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    strategy:
+      matrix:
+        php-versions: ['5.4', '5.5', '5.6', '7.0', '7.1']
+        # , '7.2', '7.3', '7.4', '8.0', '8.1'
+    steps:
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: ${{ matrix.php-versions }}
+
```
Comment (PierreRambaud) [inline .github/workflows/php.yml:17]:
```suggestion

    - uses: actions/checkout@v2

```

**Comment by PierreRambaud** [comment]
> @Progi1984 I fetch your branch and pushed it on my fork.
> 
> It fails for a very weird reason 
> 
> https://github.com/matks/decimal/runs/1935660154
> 
> ```
> # Install dependencies
> 
> 0s
> Run composer install --no-interaction
>   composer install --no-interaction
>   shell: /bin/bash -e {0}
> Composer could not find a composer.json file in /home/runner/work/decimal/decimal
> To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
> Error: Process completed with exit code 1.
> ```

Should be better with my change

**Comment by matks** [comment]
> > @Progi1984 I fetch your branch and pushed it on my fork.
> > It fails for a very weird reason
> > https://github.com/matks/decimal/runs/1935660154
> > ```
> > # Install dependencies
> > 
> > 0s
> > Run composer install --no-interaction
> >   composer install --no-interaction
> >   shell: /bin/bash -e {0}
> > Composer could not find a composer.json file in /home/runner/work/decimal/decimal
> > To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
> > Error: Process completed with exit code 1.
> > ```
> 
> Should be better with my change

There was no checkout ? Haha I missed that 🤣 shame on me

**Comment by Progi1984** [comment]
Thanks @PierreRambaud @eternoendless @matks

**Pull request merged**